test/upsert/rocksdb-cleanup/drop-source-in-cluster.td

```diff
@@ -0,0 +1,35 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ kafka-create-topic topic=upsert
+
+> CREATE CONNECTION conn
+  FOR KAFKA BROKER '${testdrive.kafka-addr}'
+
+> DROP CLUSTER IF EXISTS c1 CASCADE;
+
+> CREATE CLUSTER c1 REPLICAS (replica1 (SIZE '1'));
+
+> CREATE SOURCE upsert
+  IN CLUSTER c1
+  FROM KAFKA CONNECTION conn (TOPIC
+  'testdrive-upsert-${testdrive.seed}'
+  )
+  KEY FORMAT TEXT VALUE FORMAT TEXT
+  ENVELOPE UPSERT
+
+$ kafka-ingest format=bytes topic=upsert key-format=bytes key-terminator=:
+fish:fish
+bird:goose
+animal:whale
+
+> SELECT count(*) from upsert;
+3
+
+> DROP SOURCE upsert;
```

Comment:
Where's the code for this cleanup? Did it already land in an earlier PR? AFAICT the `remove_dir_all` added in this PR only applies on `DROP CLUSTER`, so I'm surprised the test passes!

Reply:
Yeah, this was already there as part of the rocksdb module https://github.com/MaterializeInc/materialize/blob/main/src/rocksdb/src/lib.rs#L630. So automatically handled when we just drop the source (but the cluster remains to actually process).